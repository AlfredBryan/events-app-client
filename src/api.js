import axios from "axios";

export const callApi = (endpoint, method, data = {}) => {
  return axios({
    url: endpoint,
    method,
    data
  })
    .then(response => {}) // handle response)
    .catch(error => {
      throw error;
    });
};
