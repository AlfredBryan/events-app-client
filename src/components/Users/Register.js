import React, { Component } from "react";
import axios from "axios";
import { Link } from "react-router-dom";

import "./User.css";

const apiURL = "https://events-apps.herokuapp.com/api/user/register";

class Register extends Component {
  state = {
    fullName: "",
    email: "",
    password: "",
    errorMessage: null
  };

  handleSubmit = e => {
    e.preventDefault();
    let { fullName, email, password } = this.state;

    axios
      .post(`${apiURL}`, {
        fullName,
        email,
        password
      })
      .then(res => {
        if (res.status === 200) {
          localStorage.setItem("token", res.data.token);
          this.props.history.replace("/home");
        }
      })
      .catch(error => {
        if (error) {
          this.setState({
            errorMessage: "Error: confirm details and try again"
          });
        }
      });
  };

  handleTextChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  render() {
    return (
      <React.Fragment>
        <div className="container register">
          <div className="row">
            <div className="col-md-3 register-left">
              <img
                src="https://image.ibb.co/n7oTvU/logo_white.png"
                alt="login"
              />
              <h3>Welcome.</h3>
              <Link to="/login">
                {" "}
                <button style={{ fontWeight: "bold", background: "white" }}>
                  Login
                </button>{" "}
              </Link>
              <br />
            </div>
            <div className="col-md-9 register-right">
              <h3 className="register-heading">Please Register</h3>
              <div className="row register-form">
                <div className="col-md-6">
                  <div className="form-group">
                    <input
                      type="text"
                      name="fullName"
                      id="fullName"
                      className="form-control"
                      placeholder="Full Name *"
                      value={this.state.fullName}
                      onChange={this.handleTextChange}
                    />
                  </div>
                  <div className="form-group">
                    <input
                      type="email"
                      name="email"
                      id="email"
                      className="form-control"
                      autoCapitalize="none"
                      placeholder="Email *"
                      value={this.state.email}
                      onChange={this.handleTextChange}
                    />
                  </div>
                </div>
                <div className="col-md-6">
                  <div className="form-group">
                    <input
                      type="password"
                      name="password"
                      id="password"
                      className="form-control"
                      minLength="7"
                      maxLength="20"
                      placeholder="Password *"
                      value={this.state.password}
                      onChange={this.handleTextChange}
                    />
                  </div>
                  <div className="form-group" />
                  <input
                    type="submit"
                    className="btnRegister"
                    id="submitButton"
                    value="Register"
                    onClick={this.handleSubmit}
                  />
                </div>
              </div>
            </div>
          </div>
          <p style={{ color: "white", fontSize: "20px" }}>
            {this.state.errorMessage}
          </p>
        </div>
      </React.Fragment>
    );
  }
}

export default Register;
