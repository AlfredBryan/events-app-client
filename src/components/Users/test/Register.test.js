import React from "react";
import { shallow } from "enzyme";
import Register from "../Register";
import { callApi } from "../../../api";
import axios from "axios";

describe("Register Component", () => {
  it("should render without throwing an error", () => {
    expect(
      shallow(<Register />)
        .find("form.register")
        .exists()
    ).toMatchSnapshot();
  });
  it("renders a fullName input", () => {
    expect(shallow(<Register />).find("#fullName").length).toMatchSnapshot();
  });
  it("renders a email input", () => {
    expect(shallow(<Register />).find("#email").length).toMatchSnapshot();
  });
  it("renders a password input", () => {
    expect(shallow(<Register />).find("#password").length).toMatchSnapshot();
  });
});

describe("fullName input", () => {
  it("should respond to change event and change the state of the Register Component", () => {
    const wrapper = shallow(<Register />);
    wrapper.find("#fullName").simulate("change", {
      target: { name: "email", value: "test name" }
    });

    expect(wrapper.state("fullName")).toMatchSnapshot("test name");
  });
});

describe("Email input", () => {
  it("should respond to change event and change the state of the Register Component", () => {
    const wrapper = shallow(<Register />);
    wrapper.find("#email").simulate("change", {
      target: { name: "email", value: "test@gmail.com" }
    });

    expect(wrapper.state("email")).toMatchSnapshot("test@gmail.com");
  });
});

describe("Password input", () => {
  it("should respond to change event and change the state of the Register Component", () => {
    const wrapper = shallow(<Register />);
    wrapper.find("#password").simulate("change", {
      target: { name: "password", value: "testp" }
    });

    expect(wrapper.state("password")).toEqual("testp");
  });
});
