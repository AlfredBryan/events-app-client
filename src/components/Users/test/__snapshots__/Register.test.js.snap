// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email input should respond to change event and change the state of the Register Component: test@gmail.com 1`] = `"test@gmail.com"`;

exports[`Register Component renders a email input 1`] = `1`;

exports[`Register Component renders a fullName input 1`] = `1`;

exports[`Register Component renders a password input 1`] = `1`;

exports[`Register Component should render without throwing an error 1`] = `false`;

exports[`fullName input should respond to change event and change the state of the Register Component: test name 1`] = `""`;
